### DISCUSSION TITLE
Apakah keunggulan Flutter dibandingkan bahasa lain ?

### DISCUSSION THREAD

**Discussion by aoctavia** [fleetimee/infinite-impermanence#73] [Q&A] [answered]
self explanatory

**Comment by fleetimee** [comment, marked as answer]
1. Pengembangan Aplikasi Lebih Mudah dan Cepat 
Alasan pertama Anda harus mencoba Flutter [framework adalah](https://www.niagahoster.co.id/blog/apa-itu-framework/) tersedianya fitur-fitur yang membantu proses pengembangan jadi lebih mudah dan cepat. 

Flutter menyediakan user interface widgets yang dikembangkan oleh Google dengan kualitas coding yang tinggi. Widgets ini bisa Anda kustomisasi dengan mudah, tanpa perlu menulis coding dari nol.  

Tak hanya itu, Flutter juga memiliki fitur Hot Reload yang akan menampilkan hasil coding dengan kilat. Jadi, Anda bisa bereksperimen menambahkan fitur dan memperbaiki bug lebih cepat. 

Fitur Hot Reload Flutter
Sumber: flutter.dev
Di samping itu, Anda tak perlu menguasai bahasa lain untuk membuat aplikasi di platform yang berbeda. Cukup manfaatkan widget yang disediakan Flutter, Anda bisa membuat aplikasi untuk Android dan iOS dalam sekali waktu. 

2. Custom User Interface yang Menarik
Selain memudahkan developer memuat aplikasi, Flutter juga menawarkan tampilan [user interface](https://www.niagahoster.co.id/blog/user-interface/) aplikasi yang menarik.  

Anda bisa memanfaatkan koleksi widget untuk tampilan aplikasi, seperti layout, navigasi, animasi, style tema, font, hingga pengaturan tampilan scrolling.  

Menariknya, widget tersebut bisa Anda kustomisasi sesuai dengan kebutuhan Anda. Jadi, Anda bisa membuat tampilan aplikasi yang unik dengan mudah. 

Berikut ini contoh tampilan desain aplikasi Reflectly yang memenangkan [penghargaan desain aplikasi](https://design.google/library/material-design-awards-2019/#reflectly) paling inovatif. Aplikasi ini memiliki transisi yang smooth, animasi dan gambar yang menarik, hingga tone warna yang pas. 


Seperti yang bisa Anda tebak, aplikasi tersebut menggunakan framework Flutter dalam pembuatannya. 

3. Performa seperti Aplikasi Native 
Walaupun aplikasi Flutter bersifat multiplatform, performa aplikasi yang dihasilkan tak kalah dari aplikasi native. 

Sebab, flutter mengkombinasikan kode aplikasi native dan mesin render yang canggih. Hasilnya, aplikasi tetap bisa ditampilkan dengan baik di berbagai platform. 

Tak hanya tampilan desain yang tetap menarik, performa untuk transisi, scrolling, loading data pun tetap optimal di segala platform.

Selain memang menyediakan aplikasi dengan performa optimal, Flutter juga loyal membagikan tips dan trik supaya developer mampu membuat aplikasi yang ringan. 

Cara terbaik meningkatkan performa aplikasi adalah dengan melakukan testing dan melihat halaman metrik performa. Baik itu, metrik kecepatan, memory, app size, dan daya.

Flutter juga sudah menyediakan toolsnya kok, misalnya, Performance Overlay, Widget Rebuild Tracker, DevTools Memory, dan sebagainya.  

Dengan performa yang terjaga, user experience pengguna aplikasi juga bisa meningkat. 

Baca juga: [Apa itu Aplikasi Native? Simak Penjelasan Lengkapnya!](https://www.niagahoster.co.id/blog/aplikasi-native-adalah/)

4. Biaya Pengembangan Lebih Hemat 
Seperti diketahui, Flutter adalah framework pembuatan aplikasi multiplatform. Itu artinya, proses pengembangan bisa lebih efisien. Jadi, Anda tak perlu membuat aplikasi dua kali untuk platform yang berbeda.  

Ditambah lagi dengan adanya fitur Hot Reload, UI widget, data backend, platform integration, aksesibilitas dan internasionalisasi, packages dan plugin yang memudahkan developer membuat aplikasi. Jadi, mereka bisa menghemat tenaga, waktu, dan biaya lebih banyak. 

Hal inilah yang jadi salah satu alasan banyak developer menyerbu framework Flutter. Keuntungan ini tentu sulit Anda dapatkan jika menggunakan framework lainnya yang belum menyediakan fitur tersebut.
